opentelekomcloud/services/ecs/data_source_opentelekomcloud_compute_flavor_v2.go

```diff
@@ -0,0 +1,243 @@
+				Optional: true,
+			},
+			"extra_specs": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceComputeFlavorV2Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := config.ComputeV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmterr.Errorf(errCreateV2Client, err)
+	}
+
+	var allFlavors []flavors.Flavor
+	if flavorID := d.Get("flavor_id").(string); flavorID != "" {
+		flavor, err := flavors.Get(client, flavorID).Extract()
+		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault404); ok {
+				return fmterr.Errorf("no flavor found")
+			}
+			return fmterr.Errorf("unable to retrieve OpenTelekomCloud %s flavor: %w", flavorID, err)
+		}
+
+		allFlavors = append(allFlavors, *flavor)
+	} else {
+		accessType := flavors.AllAccess
+		if isPublic, ok := d.GetOk("is_public"); ok {
+			accessType = isPublic.(flavors.AccessType)
+		}
+		listOpts := flavors.ListOpts{
+			MinDisk:    d.Get("min_disk").(int),
+			MinRAM:     d.Get("min_ram").(int),
+			AccessType: accessType,
+		}
+
+		log.Printf("[DEBUG] opentelekoncloud_compute_flavor_v2 ListOpts: %#v", listOpts)
+
+		allPages, err := flavors.ListDetail(client, listOpts).AllPages()
+		if err != nil {
+			return fmterr.Errorf("Unable to query OpenTelekomCloud flavors: %w", err)
+		}
+
+		allFlavors, err = flavors.ExtractFlavors(allPages)
+		if err != nil {
+			return fmterr.Errorf("Unable to retrieve OpenTelekomCloud flavors: %w", err)
+		}
+	}
+
+	// Loop through all flavors to find a more specific one.
+	if len(allFlavors) > 0 {
+		var filteredFlavors []flavors.Flavor
+		for _, flavor := range allFlavors {
+			if v := d.Get("name").(string); v != "" {
+				if flavor.Name != v {
+					continue
+				}
+			}
+
+			// d.GetOk is used because 0 might be a valid choice.
+			if v, ok := d.GetOk("ram"); ok {
+				if flavor.RAM != v.(int) {
+					continue
+				}
+			}
+
+			if v, ok := d.GetOk("vcpus"); ok {
+				if flavor.VCPUs != v.(int) {
+					continue
+				}
+			}
+
+			if v, ok := d.GetOk("disk"); ok {
+				if flavor.Disk != v.(int) {
+					continue
+				}
+			}
+
+			if v, ok := d.GetOk("swap"); ok {
+				if flavor.Swap != v.(int) {
+					continue
+				}
+			}
+
+			if v, ok := d.GetOk("rx_tx_factor"); ok {
+				if flavor.RxTxFactor != v.(float64) {
+					continue
+				}
+			}
+
+			es, err := flavors.ListExtraSpecs(client, flavor.ID).Extract()
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			if v, ok := d.GetOk("resource_type"); ok {
+				if resourceType, okType := es["resource_type"]; okType {
+					if resourceType != v.(string) {
+						continue
+					}
+				}
+			}
+
+			if v, ok := d.GetOk("availability_zone"); ok {
+				if azAvailability, okCond := es["cond:operation:az"]; okCond {
+					zones := strings.Split(azAvailability, ",")
+					for _, zone := range zones {
+						if zone != fmt.Sprintf("%s(normal)", v.(string)) {
+							continue
```

Comment:
Too much break/continue. Maybe move to a function? Not clear which cycle those break/continue belongs